### PR TITLE
enable profiles

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_commons/json_util.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import jsonschema
+
+
+def validate(config, schema_file) -> (bool, object):
```
Comment (Herklos) [inline octobot_commons/json_util.py:20]:
```suggestion
def validate(config, schema_file):
```

File: octobot_commons/profiles/profile.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    def __init__(self, profile_path: str, schema_path: str = None):
+        self.profile_id = None
+        self.path = profile_path
+        self.schema_path: str = schema_path or constants.PROFILE_FILE_SCHEMA
+        self.name = None
+        self.description = None
+        self.avatar = None
+        self.avatar_path = None
+
+        self.config = {}
```
Comment (Herklos) [inline octobot_commons/profiles/profile.py:59]:
```suggestion
        self.profile_id: str = None
        self.path: str = profile_path
        self.schema_path: str = schema_path or constants.PROFILE_FILE_SCHEMA
        self.name: str = None
        self.description: str = None
        self.avatar: object = None
        self.avatar_path: str = None

        self.config: dict = {}
```

File: octobot_commons/profiles/profile.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        Reads a profile from self.path
+        :return: None
+        """
+        with open(self.config_file()) as profile_file:
+            parsed_profile = json.load(profile_file)
+            self.profile_id = parsed_profile[constants.CONFIG_PROFILE][
+                constants.CONFIG_ID
+            ]
+            self.name = parsed_profile[constants.CONFIG_PROFILE][constants.CONFIG_NAME]
+            self.description = parsed_profile[constants.CONFIG_PROFILE][
```
Comment (Herklos) [inline octobot_commons/profiles/profile.py:72]: Can it be empty ? I mean by empty, not provided when empty

File: octobot_commons/profiles/profile_sharing.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import shutil
+import octobot_commons.constants as constants
+
+
+def export_profile(profile, export_path: str) -> str:
+    """
+    Exports the given profile into export_path, appends ".zip" as a file extension
+    :param profile: profile to export
+    :param export_path: export path ending with filename
+    :return: None
```
Comment (Herklos) [inline octobot_commons/profiles/profile_sharing.py:27]:
```suggestion
    :return: the exported file name
```

File: tests/profiles/test_profile.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+        _filter_fill_elements_mock.assert_not_called()
+
+        _validate_and_save_config_mock.reset_mock()
+        _filter_fill_elements_mock.reset_mock()
+        profile.config = {}
+        # things in config
+        global_config = {
+            profile.FULLY_MANAGED_ELEMENTS[0]: "plop",
+            profile.FULLY_MANAGED_ELEMENTS[1]: "plip",
+            "stuff": "plip"
```
Comment (Herklos) [inline tests/profiles/test_profile.py:63]: :smile:

File: tests/profiles/test_profile_sharing.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        # ensure all files got zipped
+        for root, dirs, files in os.walk(profile.path):
+            dir_path = os.path.join(other_profile, "specific_config") if "specific_config" in root else other_profile
+            assert all(
+                os.path.isfile(os.path.join(dir_path, f))
+                for f in files
+            )
+
+
+def test_export_profile_with_existing_file(profile):
```
Comment (Herklos) [inline tests/profiles/test_profile_sharing.py:61]: :+1:

**Review by Herklos — COMMENTED** [review]
Very good job :+1:

File: octobot_commons/json_util.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot_commons/json_util.py:20]: Absolutely !

File: octobot_commons/profiles/profile.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_commons/profiles/profile.py:72]: it should never happen but lets be cautious here, right

**Review by Herklos — APPROVED** [review]
Great job :100:

File: octobot_commons/configuration/configuration.py [hunk 7]
```diff
@@ -91,6 +92,21 @@ def select_profile(self, profile_id):
...
 
+    def remove_profile(self, profile_id: str) -> None:
+        """
+        Removes the given profile and deletes its folder on disk
+        :param profile_id: the id of the profile to remove
+        :return: None
+        """
+        profile = self.profile_by_id[profile_id]
+        if profile.read_only:
+            raise errors.ProfileRemovalError(f"{profile.name} profile can't be removed")
```
Comment (Herklos) [inline octobot_commons/configuration/configuration.py:103]: :+1:

File: tests/configuration/test_configuration.py [hunk 8]
```diff
@@ -87,6 +87,28 @@ def test_select_profile(config):
         assert config.profile is config.profile_by_id["1"]
 
 
+def test_remove_profile(config):
```
Comment (Herklos) [inline tests/configuration/test_configuration.py:90]: :+1:

**Review by Herklos — APPROVED** [review]

**Pull request merged**